src/RecordPlugin.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Http\Client\Plugin\Vcr;
+
+use Http\Client\Common\Plugin;
+use Http\Client\Plugin\Vcr\NamingStrategy\NamingStrategyInterface;
+use Http\Client\Plugin\Vcr\Recorder\RecorderInterface;
+use Http\Promise\Promise;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+final class RecordPlugin implements Plugin
+{
+    const HEADER_NAME = 'X-VCR-RECORD';
+
+    /**
+     * @var NamingStrategyInterface
+     */
+    private $namingStrategy;
+
+    /**
+     * @var RecorderInterface
+     */
+    private $recorder;
+
+    public function __construct(NamingStrategyInterface $namingStrategy, RecorderInterface $recorder)
+    {
+        $this->namingStrategy = $namingStrategy;
+        $this->recorder = $recorder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handleRequest(RequestInterface $request, callable $next, callable $first): Promise
+    {
+        $name = $this->namingStrategy->name($request);
+
+        return $next($request)->then(function (ResponseInterface $response) use ($name) {
+            if ($response->getStatusCode() < 300 && !$response->hasHeader(ReplayPlugin::HEADER_NAME)) {
```

Comment:
actually i think i would record any status code here. one might want to test a redirection or a 404 or anything. how do the other vcr out there handle this?

Reply:
Csr record everything, so I guess we shoud do the same